Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -706,6 +398,311 @@ internal struct JSONScanner {
     return Character(UnicodeScalar(UInt32(currentByte))!)
   }
 
+  // Parse the leading UInt64 from the provided utf8 bytes.
+  //
+  // This usually does a direct conversion of utf8 to UInt64.  It is
+  // called for both unquoted numbers and for numbers stored in quoted
+  // strings.  In the latter case, the caller is responsible for
+  // consuming the leading quote and verifying the trailing quote.
+  //
+  // If the number is in floating-point format, this uses a slower
+  // and less accurate approach: it identifies a substring comprising
+  // a float, and then uses Double() and UInt64() to convert that
+  // string to an unsigned intger.
+  //
+  // If it encounters a "\" backslash character, it returns a nil.  This
+  // is used by callers that are parsing quoted numbers.  See nextSInt()
+  // and nextUInt() below.
+  private func parseBareUInt(
+    source: UnsafeBufferPointer<UInt8>,
+    index: inout UnsafeBufferPointer<UInt8>.Index,
+    end: UnsafeBufferPointer<UInt8>.Index
+  ) throws -> UInt64? {
+    let start = index
+    let c = source[index]
+    source.formIndex(after: &index)
+    switch c {
+    case asciiZero: // 0
+      if index != end {
+        let after = source[index]
+        switch after {
+        case asciiZero...asciiNine: // 0...9
+          // leading '0' forbidden unless it is the only digit
+          throw JSONDecodingError.leadingZero
+        case asciiPeriod, asciiLowerE, asciiUpperE: // . e
+          // Slow path: JSON numbers can be written in floating-point notation
+          index = start
+          if let d = try parseBareDouble(source: source,
+                                         index: &index,
+                                         end: end) {
+            if let u = UInt64(exactly: d) {
+              return u
+            }
+          }
+          throw JSONDecodingError.malformedNumber
+        case asciiBackslash:
+          return nil
+        default:
+          return 0
+        }
+      }
+      return 0
+    case asciiOne...asciiNine: // 1...9
+      var n = UInt64(c - 48)
+      while index != end {
+        let digit = source[index]
+        switch digit {
+        case asciiZero...asciiNine: // 0...9
+          let val = UInt64(digit - asciiZero)
+          if n >= UInt64.max / 10 {
```

Comment:
Not sure that's right, if `n` is 3 shy of `UInt64.max` and `val` is 0-2, things are fine, but 3-9 would overflow. You likely need the /10 on the max still.